dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -92,10 +92,14 @@ static ClassDumper create(List<ClassPathEntry> entries) throws IOException {
     
     Map<String, ClassPathEntry> map = new HashMap<>();
     for (ClassPathEntry entry : entries) {
-      for (String className : entry.getFileNames()) {
-        if (!map.containsKey(className)) {
-          map.put(className, entry);
+      try {
+        for (String className : entry.getFileNames()) {
+          if (!map.containsKey(className)) {
+            map.put(className, entry);
+          }
         }
+      } catch (IOException ex) {
+        throw new IOException("Failed to read content of " + entry.getJar(), ex);
```

Comment:
Maybe this should happen inside ClassPathEntry instead

Reply:
Nice idea. Updated.